x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_with_fallback.tsx

```diff
@@ -77,6 +78,7 @@ export const useRuleWithFallback = (ruleId: string): UseRuleWithFallback => {
   }, [addError, error]);
 
   const rule = useMemo<Rule | undefined>(() => {
+    // TODO: Is this access still kosher?
     const result = isExistingRule ? ruleData : alertsData?.hits.hits[0]?._source.signal.rule;
```

Comment:
@marshallmain

Reply:
It'll work until we start using the new detection engine implementation in 7.16. If we still need the fallback, the easiest thing might be to check both `_source.kibana.alert.rule` and `_source.signal.rule` to find whichever one exists.

Reply:
Are we going to update `fetchQueryAlerts` to use the new rule registry `find` API instead of the detection engine api?

Reply:
Yea, we should, but I would consider that out of scope for this PR